p2p/src/peer_manager/tests/utils.rs

```diff
@@ -32,3 +37,21 @@ pub fn cmd_to_peer_man_msg(cmd: Command) -> (PeerId, PeerManagerMessage) {
     let msg = assert_matches_return_val!(msg, CategorizedMessage::PeerManagerMessage(msg), msg);
     (peer_id, msg)
 }
+
+pub async fn recv_command_advance_time(
+    cmd_receiver: &mut mpsc::UnboundedReceiver<Command>,
+    time_getter: &P2pBasicTestTimeGetter,
+    advance_duration: Duration,
+) -> Result<Command, mpsc::error::TryRecvError> {
+    loop {
+        match cmd_receiver.try_recv() {
+            Err(mpsc::error::TryRecvError::Empty) => {
+                time_getter.advance_time(advance_duration);
+                tokio::time::sleep(Duration::from_millis(100)).await;
```

Comment:
Would `tokio::time::advance` work here? Could save on testing time.

Reply:
I don't think so. As I understand, `tokio::time::advance` will only cause the currently running calls of `sleep` to complete. Here the purpose of `sleep` is to give tasks some real time to do the job and to prevent the loop from advancing the mocked time too often.